### ISSUE TITLE
Oxy 4.7.1/ 4.8 PHP Warning 'Undefined array key {n}' in Revisions management 

### ISSUE THREAD

**Issue by maltmann-muc** [soflyy/oxygen-bugs-and-features#3456]
**Environment**: PHP 8.1, WP 6.4.1, Oxy 4.7.1 + 4.8 RC 1
**Situation:** In Builder, saving template/post with max_revisions set 

**File and Function:**
wp-content/plugins/oxygen/component-framework/includes/revisions.php
create_revision()

**Warning and Call Stack:**
```
Warning:  Undefined array key 21 in /.../wp-content/plugins/oxygen/component-framework/includes/revisions.php on line 48
Stack trace:
  1. {main}() /.../index.php:0
  2. require() /.../index.php:17
  3. require_once() /.../wp-blog-header.php:19
  4. do_action($hook_name = 'template_redirect') /.../wp-includes/template-loader.php:13
  5. WP_Hook->do_action($args = [0 => '']) /.../wp-includes/plugin.php:517
  6. WP_Hook->apply_filters($value = '', $args = [0 => '']) /.../wp-includes/class-wp-hook.php:348
  7. ct_save_components_tree_as_post('') /.../wp-includes/class-wp-hook.php:324
  8. Oxygen_Revisions::create_revision($post_id = 9) /.../wp-content/plugins/oxygen/component-framework/includes/ajax.php:212
Warning:  Attempt to read property "meta_value" on null in /.../wp-content/plugins/oxygen/component-framework/includes/revisions.php on line 48
Stack trace:
  1. {main}() /.../index.php:0
  2. require() /.../index.php:17
  3. require_once() /.../wp-blog-header.php:19
  4. do_action($hook_name = 'template_redirect') /.../wp-includes/template-loader.php:13
  5. WP_Hook->do_action($args = [0 => '']) /.../wp-includes/plugin.php:517
  6. WP_Hook->apply_filters($value = '', $args = [0 => '']) /.../wp-includes/class-wp-hook.php:348
  7. ct_save_components_tree_as_post('') /.../wp-includes/class-wp-hook.php:324
  8. Oxygen_Revisions::create_revision($post_id = 9) /.../wp-content/plugins/oxygen/component-framework/includes/ajax.php:212

```

**Description:**
$revisions is an array, e.g. 0..24 = count 25
Loop runs for $index <= 25. $dates doesn't contain an index 25.

<img width="1864" alt="Bildschirmfoto 2023-11-26 um 09 42 08" src="https://github.com/soflyy/oxygen-bugs-and-features/assets/51136650/9760af11-fd1f-4d17-8763-d68ea5c46a83">

**Possible Solutions (not tested):**
Let loop run for $index < count($revisions) instead of <= 
 Or check $dates[$index] before accessing it.

**Comment by maltmann-muc** [comment]
Oxy 4.8: Issue persists.

**Comment by Kpudlo** [comment]
Confirmed and reported internally: https://github.com/soflyy/oxygen/issues/5225

**Comment by danner26** [comment]
Any update on this?

**Comment by maltmann-muc** [comment]
Seems to be fixed with Oxygen 4.9, according to change log, my own code review, and tests.
Thx.

**Comment by Spellhammer** [comment]
Fixed in Oxygen 4.9.
